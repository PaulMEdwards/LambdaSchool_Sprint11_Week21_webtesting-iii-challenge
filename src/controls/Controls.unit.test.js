import React from "react";
import * as rtl from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Controls from "./Controls";

test("Controls Unlocked Open", async () => {
  const controls = rtl.render(<Controls locked={false} closed={false} />);
  await controls.findByText(/Lock Gate/);
  await controls.findByText(/Close Gate/);
  expect(controls.asFragment()).toMatchSnapshot();

  const buttons = document.querySelectorAll(".toggle-btn");
  expect(buttons).toHaveLength(2);

    const controlLockUnlock = buttons[0];
    expect(controlLockUnlock).not.toBeEnabled();
    expect(controlLockUnlock.textContent).toBe("Lock Gate");

    const controlOpenClose  = buttons[1];
    expect(controlOpenClose).toBeEnabled();
    expect(controlOpenClose.textContent).toBe("Close Gate");
});

test("Controls Unlocked Closed", async () => {
  const controls = rtl.render(<Controls locked={false} closed={true} />);
  await controls.findByText(/Lock Gate/);
  await controls.findByText(/Open Gate/);
  expect(controls.asFragment()).toMatchSnapshot();

  const buttons = document.querySelectorAll(".toggle-btn");
  expect(buttons).toHaveLength(2);

    const controlLockUnlock = buttons[0];
    expect(controlLockUnlock).toBeEnabled();
    expect(controlLockUnlock.textContent).toBe("Lock Gate");

    const controlOpenClose  = buttons[1];
    expect(controlOpenClose).toBeEnabled();
    expect(controlOpenClose.textContent).toBe("Open Gate");
});

test("Controls Locked Open", async () => {
  const controls = rtl.render(<Controls locked={true} closed={false} />);
  await controls.findByText(/Unlock Gate/);
  await controls.findByText(/Close Gate/);
  expect(controls.asFragment()).toMatchSnapshot();

  const buttons = document.querySelectorAll(".toggle-btn");
  expect(buttons).toHaveLength(2);

    const controlLockUnlock = buttons[0];
    expect(controlLockUnlock).not.toBeEnabled();
    expect(controlLockUnlock.textContent).toBe("Unlock Gate");

    const controlOpenClose  = buttons[1];
    expect(controlOpenClose).not.toBeEnabled();
    expect(controlOpenClose.textContent).toBe("Close Gate");
});

test("Controls Unlocked Closed", async () => {
  const controls = rtl.render(<Controls locked={false} closed={true} />);
  await controls.findByText(/Lock Gate/);
  await controls.findByText(/Open Gate/);
  expect(controls.asFragment()).toMatchSnapshot();

  const buttons = document.querySelectorAll(".toggle-btn");
  expect(buttons).toHaveLength(2);

    const controlLockUnlock = buttons[0];
    expect(controlLockUnlock).toBeEnabled();
    expect(controlLockUnlock.textContent).toBe("Lock Gate");

    const controlOpenClose  = buttons[1];
    expect(controlOpenClose).toBeEnabled();
    expect(controlOpenClose.textContent).toBe("Open Gate");
});