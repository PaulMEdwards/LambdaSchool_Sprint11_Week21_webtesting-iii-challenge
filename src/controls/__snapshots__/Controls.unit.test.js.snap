// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls Locked Open 1`] = `
<DocumentFragment>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      disabled=""
    >
      Unlock Gate
    </button>
    <button
      class="toggle-btn"
      disabled=""
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;

exports[`Controls Unlocked Closed 1`] = `
<DocumentFragment>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Open Gate
    </button>
  </div>
</DocumentFragment>
`;

exports[`Controls Unlocked Closed 2`] = `
<DocumentFragment>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Open Gate
    </button>
  </div>
</DocumentFragment>
`;

exports[`Controls Unlocked Open 1`] = `
<DocumentFragment>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;
